### ISSUE TITLE
Pre/Post operation filters/hooks for modules/plugins to utilize

### ISSUE THREAD

**Issue by cmcculloh** [cmcculloh/GitScripts#74]
@chriscorwin @Smolations Are you guys still using admin and branch protection over at FL? I found it added a small layer of complexity both to getting gitscripts setup and configured on a new machine and the code-base in general, so, I've ripped it out of GitScripts3. However, I recognize that will be a major blocker if we ever want to fold GitScripts3 into master if you guys are actually using those features.

Basically, I'm second guessing my decision to have ripped it out. If you guys are actively using it I may either fold it back in, or implement some sort of pre/post processing filters/hooks that are extensible for GitScripts Plugins in general and implement the admin and branch protection as plugins that utilize those hooks.

**Comment by cmcculloh** [comment]
I have zero plans to do this. If someone else wants to tackle it, feel free to reopen.
